test/HeadersTest.php

```diff
@@ -400,6 +400,29 @@ public function testClone()
         $this->assertFalse($headers2->has('Bcc'));
     }
 
+    /**
+     * Problem:
+     * 1. GenericHeader loads the header in file, decodes it to utf-8
+     * 2. the Headers::get attempts to Lazy-Load "To" header class
+     * Lazyload does stringify and load in from string
+     *    $encoding = $current->getEncoding();
+     *    $headers  = $class::fromString($current->toString());
+     * However, toString does not encode comma
+     * AND To header class does split on comma!
+     *
+     * @see \Zend\Mail\Header\AbstractAddressList::fromString
+     */
+    public function testDogFood() {
+        $headers = new Mail\Headers();
+
+        $genericHeader = Mail\Header\GenericHeader::fromString('To: "=?iso-8859-1?Q?W=2C_bj=F8rn?=" <user@example.org>');
+        $this->assertEquals('"W, bjørn" <user@example.org>', $genericHeader->getFieldValue());
+
+        $headers->addHeader($genericHeader);
```

Comment:
This test case doesn't make sense given the assertions in your comments and the issue description.

You're making the case that `AbstractAddressList::fromString()` is splitting on a comma, but that `toString()` on such headers is not encoding it.

What I'm seeing here is quite different:

- You're testing the behavior of a `GenericHeader`, not one that is based on `AbstractAddressList`.
- Your assertion is that the expected behavior of `toString()` is NOT to encode the comma.

That's the crux of my confusion: the test is not setting up the conditions you describe, nor testing the expectation you describe.

Can you please clarify?